### PR TITLE
Iroh 6384 mitre attack pattern retrieval

### PR DESCRIPTION
**Pull request by marioaquino**
> Related https://github.com/advthreat/iroh/issues/6384
> Related https://github.com/advthreat/iroh/pull/6386

Adds a convenience `GET` route for retrieving MITRE attack patterns (specifically) via `external_references.url` or `external_references.external_id`.  Returns the matching MITRE attack pattern record or a 404 if no MITRE record is found.

<a name="qa">[§](#qa)</a> QA
============================

Using the `/ctia/attack-pattern/mitre/<mitre-id>` endpoint, observe these outcomes given the following conditions:

1.  Any MITRE attack pattern should be retrievable (as the `mitre-id` parameter) via either its `external_references.url` or `external_references.external_id` values.
2. Any non-MITRE attack patterns having an `external_references.url` value submitted as the `<mitre-id>` parameter will result in a 404 response.
3. For a situation where more than one version of the same MITRE attack pattern record is found, only the latest will be returned when the `external_references.external_id` value is supplied as the `<mitre-id>` parameter.

<a name="release-notes">[§](#release-notes)</a> Release Notes
=============================================================

```
intern: helper for retrieving MITRE attack patterns
```

<a name="squashed-commits">[§](#squashed-commits)</a> Squashed Commits
======================================================================

### CODE REVIEW

File: test/ctia/entity/attack_pattern/core_test.clj [hunk 1]
```diff
@@ -0,0 +1,91 @@
+(ns ctia.entity.attack-pattern.core-test
+  (:require [clojure.test :refer [deftest is testing join-fixtures use-fixtures]]
+            [ctia.entity.attack-pattern.core :as sut]
+            [ctia.test-helpers.auth :refer [all-capabilities]]
+            [ctia.test-helpers.core :as helpers]
+            [ctia.test-helpers.fake-whoami-service :as whoami-helpers]
+            [ctia.test-helpers.fixtures :as fixtures]
+            [java-time :refer [adjust days instant plus to-java-date]]
```
Comment (ereteog) [inline test/ctia/entity/attack_pattern/core_test.clj:8]: fwiw I scrolled up to check from where the time function came from.
I would use this alias:
```suggestion
            [java-time :as jt]
```

File: test/ctia/entity/attack_pattern/core_test.clj [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                                       (sut/mitre-attack-pattern services
+                                                                 identity-map
+                                                                 (get-in attack-pattern [:external_references 0 k])))
+           get-attack-pattern-by-external-id (partial get-attack-pattern-by-key :external_id)
+           get-attack-pattern-by-external-url (partial get-attack-pattern-by-key :url)]
+
+       (doseq [attack-pattern [mitre-attack-pattern-1
+                               near-duplicate-of-mitre-attack-pattern-1
+                               mitre-attack-pattern-2
+                               non-mitre-attack-pattern]]
```
Comment (ereteog) [inline test/ctia/entity/attack_pattern/core_test.clj:73]: you can add them all at once with bulk:
https://github.com/threatgrid/ctia/blob/5b5bba3e577ac5fe2a01addde53a49001896e62f/test/ctia/test_helpers/core.clj#L409

**Review by ereteog — APPROVED** [review]
great!
I confirm that it would be nice to add an end to end test with the route.

**Comment by jagrutpa** [comment]
Tested with release v 1.94.
Tested all the above test scenarios mentioned in QA Section.

No issues found hence marking this as QA Verified.

**Pull request merged**